draft-ietf-jsonpath-base.md

```diff
@@ -1510,11 +1519,11 @@ immediate contexts, which lead to the following conditions for well-typedness:
 
 {:vspace}
 As a `test-expr` in a logical expression:
-: The function's declared result type is `LogicalType`, or
-  (giving rise to conversion as per {{type-conv}}) `NodesType`.
+: The function must be either `logical-function-expr`
+  or `nodes-function-expr`.
 
 As a `comparable` in a comparison:
-: The function's declared result type is `ValueType`.
+: The function must be `value-function-expr`.
```

Comment:
This text basically becomes redundant information as the ABNF specifies it, but it may be nice for the reader to have it here as well.

Reply:
I think it would be better to leave the text on lines 1513-1517 unchanged because it is making semantic rather than syntactic statements. Also, I think it's helpful to retain the parenthetical remark on line 1514.